AK/URLParser.cpp

```diff
@@ -382,8 +564,22 @@ static Optional<DeprecatedString> parse_host(StringView input, bool is_not_speci
         }
     }
 
-    auto ipv4_host = parse_ipv4_address(ascii_domain);
-    return ipv4_host;
+    // 8. If asciiDomain ends in a number, then return the result of IPv4 parsing asciiDomain.
+    if (is_ascii_digit(ascii_domain[ascii_domain.length() - 1])) {
```

Comment:
Is it possible for `input` to have been an empty string?

Reply:
No, step 3. of the spec has an assertion that input is not empty (which is later copied to asciiDomain